sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -125,7 +131,6 @@ def _hdbscan_brute(
         distance_matrix = pairwise_distances(
             X, metric=metric, n_jobs=n_jobs, **metric_params
         )
-    distance_matrix /= alpha
```

Comment:
This is technically a bit out of scope for this PR since it lives in `_hdbscan_brute`, however upon investigation I noticed:
1. We do not have any tests for the behavior of the `alpha` parameter.
2. It is unlikely to be used.
3. It is _unadvised_ to use.
4. It currently has no effect in `_hdbscan_brute` (carryover bug?).

Hence I opted to remove it from `_linkage.pyx`. Since it was essentially only used in `mst_from_data_matrix` and the ineffective code above, I opted to remove it from `HDBSCAN` entirely. The above change does not affect any HDBSCAN behavior in a meaningful way.

Reply:
Thanks for spotting this. Do you think it is possible to extract it in another PR? This way, we can discuss the problem independently from this PR which can be merged in the meantime.

Reply:
I agree removing `alpha` should be in it's own PR. This changes the public API and could become a blocker for including HDBSCAN.